amgut/lib/data_access/ag_data_access.py

```diff
@@ -969,3 +969,303 @@ def is_deposited_ebi(self, barcode):
             sql = "SELECT deposited FROM ag.ag_kit_barcodes WHERE barcode = %s"
             TRN.add(sql, [barcode])
             return TRN.execute_fetchlast()
+
+    # following are DB access functions only used for unit testing:
+
+    def get_random_supplied_kit_id_scanned_unconsented(self):
```

Comment:
We might be able to subclass it - but perhaps that's a bit outside the scope of this PR